streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/AppMaster.scala

```diff
@@ -83,26 +89,27 @@ class AppMaster(appContext : AppMasterContext, app : AppDescription)  extends Ap
   actorSystem.eventStream.subscribe(historyMetricsService, classOf[MetricType])
 
   override def receive : Receive =
-    taskMessageHandler orElse
+      taskMessageHandler orElse
       executorMessageHandler orElse
       recover orElse
-      appMasterInfoService orElse
+      appMasterService orElse
       ActorUtil.defaultMsgHandler(self)
 
   def taskMessageHandler: Receive = {
     case clock: UpdateClock =>
-      taskManager forward clock
+      clockService.foreach(_ forward clock)
+    case clock: GetUpstreamMinClock =>
+      clockService.foreach(_ forward clock)
     case GetLatestMinClock =>
-      taskManager forward GetLatestMinClock
+      clockService.foreach(_ forward GetLatestMinClock)
```

Comment:
It seems that with this change `TaskManager` doesn't need `ClockService` any more ?